### PR TITLE
CLN: post-post numpy bump

### PR DESCRIPTION
**Pull request by h-vetinari**
Even after #23062 and #23796, I still keep finding occurrences of old numpy references in the code. Gotta catch 'em all...

### CODE REVIEW

**Review by mroeschke — APPROVED** [review]

File: pandas/_libs/tslibs/timedeltas.pyx [hunk 1]
```diff
@@ -161,9 +161,6 @@ cpdef convert_to_timedelta64(object ts, object unit):
         - None/NaT
 
     Return an ns based int64
-
-    # kludgy here until we have a timedelta scalar
-    # handle the numpy < 1.7 case
```
Comment (TomAugspurger) [inline pandas/_libs/tslibs/timedeltas.pyx]: Do you know what section of code this was referring to?

**Review by TomAugspurger — APPROVED** [review]
Just one tiny question, though this should be good to go.

**Comment by jreback** [comment]
lgtm. ex @TomAugspurger question

File: pandas/_libs/tslibs/timedeltas.pyx (hunk 1 above)
Comment (h-vetinari) [reply pandas/_libs/tslibs/timedeltas.pyx]: @TomAugspurger 
I don't see something particularly kludge-y either. Also, there is a timedelta scalar nowadays. It just seems like a severely outdated comment.

**Review by h-vetinari — COMMENTED** [review]
@TomAugspurger @jreback 
Responded in comment

**Comment by TomAugspurger** [comment]
Agreed, thanks.

**Pull request merged**